### DISCUSSION TITLE
Trying to unregister user from transport not working using GoJara (can't find any documentation to fix it)

### DISCUSSION THREAD

**Discussion by Knezzen** [SpectrumIM/spectrum2#447] [General]
Hi!

I'm very aware of this being a GoJara and not a Spectrum2 issue, but I can't find any Spectrum2 documentation to be able to fix it.

Sending a message like this to the Spectrum2 instance:
`</message> 
    <body>unregister user@domain.suffix</body> 
<message from="gojaraadmin@domain.suffix" to="facebook.domain.suffix" id="unregister" type="chat"> `

Gives me a reply like this from the transport instance:
`<message from="facebook.domain.suffix" id="unregister" to="gojaraadmin@domain.suffix" type="chat"><body>Error: This command can be executed only in user context.</body></message>`

Could someone point me to some documentation on how "unregistered" should be implemented, or is this a Spectrum2 problem after all?

**Comment by vitalyster** [comment]
You should set Gojara JID as admin JID in transport config, like this:
```
[service]
admin_jid=gojaraadmin@domain.suffix
```

It is documented here - https://discourse.igniterealtime.org/t/how-to-run-spectrum-2-with-openfire-gojara-plugin-guide/64624

**Reply by Knezzen** [reply]
That has been done according to that guide and the GoJara web interface displays everything else correctly, it's just that "unregister" doesn't work.

**Comment by Knezzen** [comment]
With
```
[service]
admin_jid=gojaraadmin@domain.suffix
```
Set in the transport configuration file, I can't unregister users manually from the command line (using spectrum2_manager) or using the web interface. Is this by design? I get the same error message as above (Error: This command can be executed only in user context.) regardless. 

The command I run is as follows:
`spectrum2_manager transport.domain.suffix unregister username`

**Reply by vitalyster** [reply]
I just checked the code and it seems like this form of the unregister command is commented out. It should work only from "user context", that means user can unregister himself by sending "unregister" to transport jid (or use registration interface in service discovery)

**Reply by Knezzen** [reply]
Aha. This causes issues when users add transports with the incorrect login information and the transport is not added to the Roster on some clients. No one is able to unregister the account, not even the administrator of the server (me in this instance) since most clients doesn't think that transport X is registered if there's no transport in the roster.

I'm highlighting this issue since I have a list of 11 users registered on my server with transports that are in this "limbo mode".
A fix (not commenting out the feature) would be highly appreciated.

**Reply by vitalyster** [reply]
@Knezzen you can delete registrations directly from transport database, `users` table

**Reply by Knezzen** [reply]
@vitalyster Thanks! Any reason to keep it commented out?

**Reply by vitalyster** [reply]
I think it needs time to work on it and expand command to be compatible with both admin and user context - https://github.com/SpectrumIM/spectrum2/blob/a6ab83d7c51b08c2f8dd241fa261a1d81dcbafa3/libtransport/AdminInterface.cpp#L762 - PRs welcomed!
